src/mlpack/core/data/load.hpp

```diff
@@ -69,9 +69,10 @@ namespace data /** Functions to load and save matrices and models. */ {
  * @param inputLoadType Used to determine the type of file to load (default arma::auto_detect).
  * @return Boolean value indicating success or failure of load.
  */
-template<typename eT>
-bool Load(const std::string& filename,
-          arma::Mat<eT>& matrix,
+template<typename MatType>
+typename std::enable_if<std::is_same<MatType, arma::Mat<typename MatType::elem_type>>::value || std::is_same<MatType, arma::Mat<typename MatType::elem_type>>::value, bool>::type
```

Comment:
I'm not sure I understand why the `std::is_same<>` is necessary here.  Actually the compiler should already do something equivalent: it will only select this overload if the second parameter matches `template<typename eT> arma::Mat<eT>` for some `eT`.  If we want to add support for loading bandicoot matrices later, we would probably want to just add another overload for `coot::Mat<eT>`, or if the function calls are the same, then we can templatize to `MatType`.  Let me know if I overlooked something.

Reply:
Agreed:

```
bool Load(const std::string& filename, arma::Mat<eT>& matrix, ...) {}
bool Load(const std::string& filename, coot::Mat<eT>& matrix, ...) {}
```

should build just fine, also we might want to put all this into some `ifdef` as well, so we can still use `arma::Mat` without having bandicoot.